esphome/components/ade7953_i2c/ade7953_i2c.cpp

```diff
@@ -0,0 +1,81 @@
+#include "ade7953_i2c.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+
+namespace esphome {
+namespace ade7953_i2c {
+
+static const char *const TAG = "ade7953";
+
+void AdE7953I2c::dump_config() {
+  ESP_LOGCONFIG(TAG, "ADE7953_i2c:");
+  LOG_I2C_DEVICE(this);
+  ade7953_base::ADE7953::dump_config();
+}
+bool AdE7953I2c::ade_write_8(uint16_t reg, uint8_t value) {
+  std::vector<uint8_t> data;
+  data.push_back(reg >> 8);
+  data.push_back(reg >> 0);
+  data.push_back(value);
+  return write(data.data(), data.size()) != i2c::ERROR_OK;
+}
+bool AdE7953I2c::ade_write_16(uint16_t reg, uint16_t value) {
+  std::vector<uint8_t> data;
+  data.push_back(reg >> 8);
+  data.push_back(reg >> 0);
+  data.push_back(value >> 8);
+  data.push_back(value >> 0);
+  return write(data.data(), data.size()) != i2c::ERROR_OK;
+}
+bool AdE7953I2c::ade_write_32(uint16_t reg, uint32_t value) {
+  std::vector<uint8_t> data;
```

Comment:
Since we know the future size we can allocate it once at the start
```suggestion
  std::vector<uint8_t> data(6);
```